python/ray/serve/_private/http_state.py

```diff
@@ -293,6 +387,33 @@ def _stop_proxies_if_needed(self) -> bool:
             proxy = self._proxy_states.pop(node_id)
             ray.kill(proxy.actor_handle, no_restart=True)
 
+    def _restart_unhealthy_proxies(self):
+        """Kill and restart unhealthy proxies.
+
+        Iterate through all proxies and check the status of them. Shutdown the unhealthy
+        proxies and restart them.
+        """
+        for node_id, node_ip_address in self._get_target_nodes():
+            proxy_state = self._proxy_states[node_id]
+            if proxy_state.status == HTTPProxyStatus.UNHEALTHY:
```

Comment:
To create new proxy it requires variables currently only exist on `HTTPState`. It will be quite a bit refactor if we want to move all those logics into `HTTPProxyState`. I would prefer to keep this part as is and possibly do this refactor in a followup in the future if no strong preference😅

Reply:
Talked with Ed, we might also want to rename those classes and possibly refactor the common behavior for this and the deployment state. Will keep the logical changes and refactor changes separate for now.